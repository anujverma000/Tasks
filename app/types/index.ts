export interface TodoCategoryProps {
  title: string;
  description: string;
  taskCount: number;
  id: string;
}

export interface TodoProps {
  id: string;
  title?: string;
  description?: string;
  completed: boolean;
  category?: TodoCategoryProps;
}

export type NotesProps = {
  title: string;
  description: string;
};
