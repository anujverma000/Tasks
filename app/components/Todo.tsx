import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {TodoProps} from '../types';

const styles = StyleSheet.create({
  container: {
    paddingVertical: 8,
    justifyContent: 'center',
  },
  title: {
    fontSize: 16,
    color: '#8C8080',
  },
  descripton: {
    fontSize: 14,
    color: '#aba2a3',
  },
  completed: {
    textDecorationLine: 'line-through',
    textDecorationStyle: 'solid',
  },
});

const Todo = (todo: TodoProps) => {
  return (
    <View style={styles.container}>
      <Text style={[styles.title, todo.completed ? styles.completed : {}]}>
        {todo.title}
      </Text>
      {todo.description ? (
        <Text
          style={[styles.descripton, todo.completed ? styles.completed : {}]}>
          {todo.description}
        </Text>
      ) : null}
    </View>
  );
};

export default Todo;
