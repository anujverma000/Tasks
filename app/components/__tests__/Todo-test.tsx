import React from 'react';
import renderer from 'react-test-renderer';
import {TodoCategoryProps, TodoProps} from '../../types';
import Todo from '../Todo';

it('Todo renders correctly', () => {
  const category: TodoCategoryProps = {
    title: 'category Title',
    description: 'category description',
    id: 'category-id',
  };

  const props: TodoProps = {
    id: '1234-test',
    title: 'Test Title',
    description: 'Test Description',
    completed: false,
    category: category,
  };

  const tree = renderer.create(<Todo {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
