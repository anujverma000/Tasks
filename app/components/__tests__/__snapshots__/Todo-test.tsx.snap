// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#8C8080",
          "fontSize": 16,
        },
        Object {},
      ]
    }
  >
    Test Title
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#aba2a3",
          "fontSize": 14,
        },
        Object {},
      ]
    }
  >
    Test Description
  </Text>
</View>
`;
